Source/Core/Core/HW/GCMemcard/GCMemcard.h

```diff
@@ -79,6 +80,29 @@ enum class GCMemcardRemoveFileRetVal
   DELETE_FAIL,
 };
 
+enum class GCMemcardValidityIssues
+{
+  FAILED_TO_OPEN,
+  IO_ERROR,
+  INVALID_CARD_SIZE,
+  INVALID_CHECKSUM,
+  FREE_BLOCK_MISMATCH,
+  DATA_IN_UNUSED_AREA,
+  COUNT
+};
+
+class GCMemcardErrorCode
+{
+private:
```

Comment:
Public interface should come before the private interface. Existing instances where this isn't the case is due to code existing before the coding guideline was introduced.